shell/main.js

```diff
@@ -34,8 +32,17 @@ if (Services.appinfo.OS === 'Darwin') {
 }
 
 const url = Runtime.commandLineArgs[0] || Runtime.packageJSON.mainURL || 'index.html';
-const argument = Cc['@mozilla.org/supports-string;1'].createInstance(Ci.nsISupportsString);
-argument.data = url;
+const shellUrl = `chrome://app/content/shell.xul`;
+
+// // We should be able to use window.open here, but we're using the hidden window
+// // as our window global object, and calling window.open on it throws:
+// //   JavaScript error: …nsPrompter.js, line 350: NS_ERROR_NOT_AVAILABLE:
+// //   Cannot call openModalWindow on a hidden window
+// //
+// // window.open(shellUrl, '_blank', windowFeatures);
+// Services.ww.openWindow(null, shellUrl, '_blank', windowFeatures, null);
 
-// TODO: report error if URL isn't found.
-Services.ww.openWindow(null, SHELL_URL, '_blank', WINDOW_FEATURES, argument);
+// Keep messing around with using window.open to open the window.
+const shell = window.open(shellUrl, '_blank', windowFeatures);
+dump(`shell: ${shell}\n`);
+shell.addEventListener('load', event => shell.loadURL(url), false);
```

Comment:
Indeed, done in 9159731. Note that the project's [.eslintrc.js file](https://github.com/mozilla/qbrt/blob/master/.eslintrc.js) sets [no-unused-vars](http://eslint.org/docs/rules/no-unused-vars), but that setting's configuration disables checking for unused parameters via `args: 'none'`:

```js
    // Some of our function definitions are XPCOM interface implementations,
    // and they don't always use all the arguments defined by the interface,
    // but it's still useful to declare their parameters, to make the interface
    // specification more obvious, so we ignore their nonuse via args: none.
    //
    'no-unused-vars': ['error', { varsIgnorePattern: 'Cc|Ci|Cr|Cu', args: 'none' }],
```

If we changed that to the default value `args: 'after-used'`, then we'd catch issues like this one (but have to change those XPCOM interface implementations), which is arguably preferable.